packages/legend-extension-external-store-service/src/components/ESService_ServiceStoreEmbeddedData.tsx

```diff
@@ -0,0 +1,997 @@
+  serviceStore_stringValuePattern_setExpectedValue,
+} from '../stores/studio/ESService_GraphModifierHelper';
+import {
+  type ServiceStoreEmbeddedDataState,
+  ServiceStubMappingState,
+  SERVICE_REQUEST_PATTERN_TAB_TYPE,
+  SERVICE_STUB_MAPPING_TAB_TYPE,
+} from '../stores/studio/ESService_ServiceStoreEmbeddedDataEditorState';
+
+export type StringValuePatternOption = {
+  value: string;
+  label: string;
+};
+
+export const QueryParamsEditor = observer(
+  (props: {
+    serviceRequestPattern: ServiceRequestPattern;
+    isReadOnly: boolean;
+  }) => {
+    const { serviceRequestPattern, isReadOnly } = props;
+    const stringValuePatternOptions = Object.values(
+      V1_StringValuePatternType,
+    ).map((stringValuePatternType) => ({
+      value: stringValuePatternType,
+      label: stringValuePatternType,
+    }));
+    const onStringValuePatternChange = (
+      val: StringValuePatternOption,
+      key: string,
+    ): void => {
+      let stringValuePattern: StringValuePattern;
+      if (val.value === V1_StringValuePatternType.EQUAL_TO_JSON_PATTERN) {
+        stringValuePattern = new EqualToJsonPattern();
+        serviceStore_stringValuePattern_setExpectedValue(
+          stringValuePattern,
+          '',
+        );
+        serviceStore_serviceRequestPattern_updateQueryParamValue(
+          serviceRequestPattern,
+          key,
+          stringValuePattern,
+        );
+      } else {
+        stringValuePattern = new EqualToPattern();
+        serviceStore_stringValuePattern_setExpectedValue(
+          stringValuePattern,
+          '',
+        );
+        serviceStore_serviceRequestPattern_updateQueryParamValue(
+          serviceRequestPattern,
+          key,
+          stringValuePattern,
+        );
+      }
+    };
+    const addQueryParam = (): void => {
+      const equalToPattern = new EqualToPattern();
+      serviceStore_stringValuePattern_setExpectedValue(equalToPattern, '');
+      serviceStore_serviceRequestPattern_addQueryParams(
+        serviceRequestPattern,
+        '',
+        equalToPattern,
+      );
+    };
+
+    return (
+      <div className="query-params-editor">
+        <div className="query-params-editor__header">
+          <div className="query-params-editor__header__title">
+            <div className="query-params-editor__header__title__label">
+              Query Params
+            </div>
+          </div>
+          <div className="query-params-editor__header__actions">
+            <button
+              className="query-params-editor__header__action"
+              onClick={addQueryParam}
+              disabled={isReadOnly}
+              tabIndex={-1}
+              title={'Add query param'}
+            >
+              <PlusIcon />
+            </button>
+          </div>
+        </div>
+        <div className="query-params-editor__content">
+          {Array.from(serviceRequestPattern.queryParams?.values() ?? []).map(
```

Comment:
since you are using both keys and values in this map, could you keep it cleaner and use `.entries()` instead of `.values()`